crates/turborepo-ci/src/lib.rs

```diff
@@ -0,0 +1,196 @@
+#![deny(clippy::all)]
+
+mod vendors;
+
+use std::{env, sync::OnceLock};
+
+use crate::vendors::get_vendors;
+pub use crate::vendors::Vendor;
+
+static IS_CI: OnceLock<bool> = OnceLock::new();
+
+pub fn is_ci() -> bool {
+    *IS_CI.get_or_init(|| {
+        // We purposefully don't do `is_err()` because the Go version
+        // returns false for both an unset env variable
+        // and an env variable set to the empty string.
+        let build_id = env::var("BUILD_ID").unwrap_or_default();
+        let build_number = env::var("BUILD_NUMBER").unwrap_or_default();
+        let ci = env::var("CI").unwrap_or_default();
+        let ci_app_id = env::var("CI_APP_ID").unwrap_or_default();
+        let ci_build_id = env::var("CI_BUILD_ID").unwrap_or_default();
+        let ci_build_number = env::var("CI_BUILD_NUMBER").unwrap_or_default();
+        let ci_name = env::var("CI_NAME").unwrap_or_default();
+        let continuous_integration = env::var("CONTINUOUS_INTEGRATION").unwrap_or_default();
+        let run_id = env::var("RUN_ID").unwrap_or_default();
+        let teamcity_version = env::var("TEAMCITY_VERSION").unwrap_or_default();
+
+        !build_id.is_empty()
+            || !build_number.is_empty()
+            || !ci.is_empty()
+            || !ci_app_id.is_empty()
+            || !ci_build_id.is_empty()
+            || !ci_build_number.is_empty()
+            || !ci_name.is_empty()
+            || !continuous_integration.is_empty()
+            || !run_id.is_empty()
+            || !teamcity_version.is_empty()
```

Comment:
We can keep the short circuiting behavior of the Go code if we hold off on querying all of the env vars we care about
```suggestion

const CI_ENV_VARS: &[&str] = [
    "BUILD_ID",
    "BUILD_NUMBER",
    "CI",
    "CI_APP_ID",
    "CI_BUILD_ID",
    "CI_BUILD_NUMBER",
    "CI_NAME",
    "CONTINUOUS_INTEGRATION",
    "RUN_ID",
    "TEAMCITY_VERSION",
]
.as_slice();

pub fn is_ci() -> bool {
    *IS_CI.get_or_init(|| {
        // We purposefully don't do `is_err()` because the Go version
        // returns false for both an unset env variable
        // and an env variable set to the empty string.
        CI_ENV_VARS
            .iter()
            .any(|env_var| !env::var(env_var).unwrap_or_default().is_empty())
```